azurerm/resource_arm_mssql_server_blob_extended_auditing_policies.go

```diff
@@ -0,0 +1,240 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/2017-03-01-preview/sql"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	uuid "github.com/satori/go.uuid"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMsSqlServerBlobExtendedAuditingPolicies() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMsSqlServerBlobExtendedAuditingPoliciesCreateUpdate,
+		Read:   resourceArmMsSqlServerBlobExtendedAuditingPoliciesRead,
+		Update: resourceArmMsSqlServerBlobExtendedAuditingPoliciesCreateUpdate,
+		Delete: resourceArmMsSqlServerBlobExtendedAuditingPoliciesDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateMsSqlServerName,
+			},
+
+			"state": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Enabled", "Disabled"}, false),
+			},
+
+			"storage_endpoint": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"storage_account_access_key": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"retention_days": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"audit_actions_and_groups": {
+				Type:     schema.TypeString,
```

Comment:
since this is a List - to make this a better UX this wants to be a Set of Strings which we convert to/from a CSV

Reply:
OK. I have made the corresponding changes in new PR #5036